### PR TITLE
Makefile for automatically generating OM client code

### PR DESCRIPTION
**Pull request by cdoron**
as well as datacatalog server code.
Currently, one can only run:
    make generate-code patch

Signed-off-by: Doron Chen <cdoron@il.ibm.com>

### CODE REVIEW

**Comment by cdoron** [comment]
This PR includes both the original swagger json for OM, and a slightly amended swagger json which I use to fix auto-generated client code.
We use openapi to generate both server and client code for the data catalog. The server code is the backbone of the connector. The client code is to help with connection information which may include unexpected fields, such as "s3" (trick suggested by Rohith)

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	docker run --rm \
+           -v ${PWD}:/local \
+           -u "${USER_ID}:${GROUP_ID}" \
+           openapitools/openapi-generator-cli generate -g go \
+           --git-user-id=${GIT_USER_ID} \
+           --git-repo-id=${GIT_REPO_ID_CLIENT} \
+           -o /local/client \
+           -i /local/client-swagger/swagger.json
+
+patch:
```
Comment (roytman) [inline Makefile:71]: why not something like ` sed -i '/github.com\/gorilla\/mux/d' api/go/api_default.go` ?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+TMP_FILE = tmpfile.tmp
+
+IMG := ${DOCKER_HOSTNAME}/${DOCKER_NAMESPACE}/${DOCKER_NAME}:${DOCKER_TAG}
+export HELM_EXPERIMENTAL_OCI=1
+
+
+all: generate-code patch
+
+.PHONY: compile
```
Comment (roytman) [inline Makefile:24]: Please comment out/remove the unused rules. we can add them gradually with relevant code

**Review by roytman — COMMENTED** [review]
2 small comments

File: Makefile (hunk 2 above)
Comment (cdoron) [reply Makefile]: done

**Review by roytman — APPROVED** [review]

**Pull request merged**